presto-cli/src/main/java/io/prestosql/cli/TableNameCompleter.java

```diff
@@ -58,10 +58,18 @@ public TableNameCompleter(QueryRunner queryRunner)
 
     private List<String> listTables(String schemaName)
     {
-        return queryMetadata(format("SELECT table_name FROM information_schema.tables WHERE table_schema = '%s'", schemaName));
+        String[] array = schemaName.split("\\.");
+        if (array.length == 1) {
+            return queryMetadata(format("SELECT table_schema || '.' || table_name FROM %s.information_schema.tables", array[0]));
+        }
+        else if (array.length == 2) {
+            return queryMetadata(format("SELECT table_name FROM information_schema.tables WHERE table_schema = '%s'", array[1]));
+        }
+        else {
+            return ImmutableList.of();
+        }
```

Comment:
nit: We can replace `else if` to `else` and remove last `else` in this case since there's `return` in each block. 

As you know, if we specify the wrong catalog or omit it, the first query become like this and it will always fail.
```sql
SELECT table_schema || '.' || table_name FROM catalog_not_exist.information_schema.tables
SELECT table_schema || '.' || table_name FROM .information_schema.tables
```

Reply:
> nit: We can replace else if to else and remove last else in this case since there's return in each block.

If catalog is omitted, enters the last `else` block.

>  if we specify the wrong catalog or omit it, the first query become like this and it will always fail.

You are right if a wrong catalog is specified. It might have to be validated as a correct catalog name.

Reply:
> If catalog is omitted, enters the last else block.

This is not correct. If catalog is omitted, it will enters the first if block since `schemaName` is empty string in this case and `"".split("\\.").length` is 0.

Reply:
schema is not included in the argument if catalog is not specified.